components/prism-factor.js

```diff
@@ -0,0 +1,370 @@
+(function (Prism) {
+
+	var comment_inside = {
+		function: {
+			pattern: /(\b)(?:TODO|FIX|FIXME|NOTE|BUG|XX+|HACK|\?\?+|!!+)(?=\b|:)/,
```

Comment:
1. `\b` is zero-width, so you don't need the lookbehind, because it will only ever capture the empty string.
2. `\?{2,}|!{2,}` might be more readable? This is just preference, so if you like the current version better, just ignore this nit.

Reply:
thanks! the theme here is that JS RegExp supports a lot more features than expected -- i thought `{n,}` was a Perl RE extension

Reply:
Well, [JS Regex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) doesn't have a lot of features but it does have all a few. Though you can't use all of them because we still want to support older browsers (stuff like RE lookbehinds, Unicode properties, and named capturing groups are all too new).